src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -67,7 +68,19 @@ public class RequestExecutor : IDisposable
 
         private ServerNode _topologyTakenFromNode;
 
-        public HttpClient HttpClient { get; }
+        private HttpClient _httpClient;
+
+        public HttpClient HttpClient
+        {
+            get
+            {
+                var httpClient = _httpClient;
+                if (httpClient != null)
+                    return httpClient;
+
+                return _httpClient = GetHttpClient();
```

Comment:
Maybe it is overkill, but does it have to be more thread-safe?

```
var newClient = GetHttpClient();
var lastKnownClient = Interlocked.CompareExchange(ref _httpClient, newClient , httpClient );
if(  lastKnownClient == httpClient )
    return newClient ;

return lastKnownClient;
```

Reply:
That would make you do a cross thread operation each time, why?